assets/src/edit-story/components/inspector/document/publish/index.js

```diff
@@ -105,28 +100,54 @@ function PublishPanel() {
     actions: { updateStory },
   } = useStory();
 
+  const {
+    actions: { showSidebarAt, hideSidebar },
+    state: { hasSidebar },
+  } = useSidebar();
+
   const { timeFormat } = useConfig();
 
-  const [showDatePicker, setShowDatePicker] = useState(false);
-  const dateTimeNode = useRef();
   const dateFieldNode = useRef();
 
-  useKeyDownEffect(dateFieldNode, { key: ['space', 'enter'] }, () => {
-    setShowDatePicker((val) => !val);
-  });
-
-  useFocusOut(dateTimeNode, () => setShowDatePicker(false), [showDatePicker]);
+  useKeyDownEffect(
+    dateFieldNode,
+    { key: ['space', 'enter'] },
+    () => {
+      if (hasSidebar) {
+        hideSidebar();
+      } else {
+        handleOpenCalendar();
+      }
+    },
+    [hasSidebar]
+  );
 
   const handleDateChange = useCallback(
     (value, close = false) => {
-      if (close && showDatePicker) {
-        setShowDatePicker(false);
+      if (close) {
```

Comment:
The date picker should be dismissible using the esc key and when pressed, keyboard focus should be returned to where it came from (the date form field in the inspector).